websocketOutportDriver/websocketOutportDriverNodePart.go

```diff
@@ -58,22 +59,38 @@ func NewWebsocketOutportDriverNodePart(args WebsocketOutportDriverNodePartArgs)
 }
 
 // SaveBlock will send the provided block saving arguments within the websocket
-func (o *websocketOutportDriverNodePart) SaveBlock(args *indexer.ArgsSaveBlockData) error {
-	return o.handleAction(args, outportSenderData.OperationSaveBlock)
+func (o *websocketOutportDriverNodePart) SaveBlock(args *outport.ArgsSaveBlockData) error {
+	var headerType string
+	if !check.IfNil(args.Header) {
+		headerType = reflect.TypeOf(args.Header).String()
+	}
+
+	argsSaveBlock := outportSenderData.ArgsSaveBlock{
+		HeaderType:        outportSenderData.HeaderType(headerType),
+		ArgsSaveBlockData: prepareArgsSaveBlock(args),
+	}
+
+	return o.handleAction(argsSaveBlock, outportSenderData.OperationSaveBlock)
 }
 
 // RevertIndexedBlock will handle the action of reverting the indexed block
 func (o *websocketOutportDriverNodePart) RevertIndexedBlock(header data.HeaderHandler, body data.BodyHandler) error {
+	var headerType string
+	if !check.IfNil(header) {
+		headerType = reflect.TypeOf(header).String()
+	}
```

Comment:
what happens in the case that the header is nil? might return an error

Reply:
It will throw an error on the client side because the header type will be unknown. 
I think it is ok as it is.